### DISCUSSION TITLE
Getting road type breakdown of a pedestrian route - e.g. side streets, path, main roads, etc.?

### DISCUSSION THREAD

**Discussion by juicetin** [valhalla/valhalla#4906] [Q&A] [answered]
Hi!

I came across a bit of Valhalla documentation that describes mapbox as a hosted version of Valhalla. On websites like https://routeshuffle.com/planner and https://trailrouter.com, I noticed that it can provide a breakdown of the pedestrian route like side streets, path, main roads, stairs, cycleway, etc.

I was digging through both the documentation, and the valhalla code itself and couldn't find this information being surface in the Valhalla APIs - is this a feature that doesn't live in the Valhalla API itself, but is an addition provided by Mapbox only?

Thanks in advance!

**Comment by chrstnbwnkl** [comment, marked as answer]
You can use `/trace_attributes` for that, which gives you all kinds of edge attributes that valhalla stores internally, including road class, use, surface type, etc. Here's the documentation: https://valhalla.github.io/valhalla/api/map-matching/api-reference/#attribute-filters-trace_attributes-only

**Reply by juicetin** [reply]
Thank you so much! I didn't look through the docs well enough, obviously :)
